torch/csrc/jit/codegen/cuda/executor_launch_params.h

```diff
@@ -125,6 +131,7 @@ class TORCH_CUDA_CU_API LaunchParams {
   int64_t bdimz_ = UNINITIALIZED_VAL;
 
   int64_t smem_ = 0;
+  int64_t nregisters_ = UNINITIALIZED_VAL;
```

Comment:
I don't think this is right as the register count is not a launch parameter. I think it should be an optional property of `Kernel`.

Reply:
I was thinking put is in `CompileOptions`. But then I saw the comments to move it to launch params. Here you suggest we move it to "class TORCH_CUDA_CU_API Kernel final : public Fusion " ?
`// TODO: Should this actually be in launch params?
struct TORCH_CUDA_CU_API CompileOptions {
  c10::Device device = c10::Device(c10::DeviceType::CUDA, 0);
  KernelIndexMode index_mode = KernelIndexMode::INT64;
};`

Reply:
Right now, `CompileOptions` is actually a property of `Kernel`. When we lower a `Fusion` to a `Kernel`, we use tensor sizes if available to determine if `int` is enough for indexing. The comment doesn't make sense to me.

I don't remember if a single `Kernel` is recompiled multiple times with different configurations, but assuming it isn't, a `Kernel` has only one compiled binary. So, right now, I think it's safe to say the max register count is a `Kernel` property. However, in the future, we may want to do some auto-tuning of different max register counts with the same `Kernel`.

In any case, I think it would make most sense to add a new data field in `HeuristicParams` to represent the max register count (or occupancy ratio) if available, and then pass it to `compileFusion` in the same as a `LaunchParams` is passed to it.

Reply:
refactored.

Reply:
I thought Kernel gets recompiled due to register count to meet the high water mark, but would have to re-read the code. I think it's fine to throw this in the launch parameters just to port around, but I also think it was fine where it was. I wonder if I was the one that left the comment.

Reply:
I already moved maxrregcount to HeuristicParams. Forgot to delete it here in LaunchParams. will update.